File: Wheel_Deflection/MATLAB/Static Radial Compression Test/getUsefulMarkers.m
% Get the indecies of the columns that pertain to the specific markers
% MTS head, Top of the rim, and center of the axel. Could be added upon 
% to include other points of interest.

% Input: 
% mocap_synced = trimmed data from the motion capture
% I = index of the MTS head column
% RimTop = the index of the RimTop y-axis indicator
% Center = the index of the Center y-axis indicator

% Output: 
% mo_time = the time column from mocap_synced
% m_head = the columns pertaining to the head indicator
% m_rim_top = the columns pertaining to the rim top indicator
% m_axel = the columns pertaining to the axel indicator

function [mo_time, m_head, m_rim_top, m_axle] = getUsefulMarkers(mocap_synced, I, RimTop, Center)

    mo_time = mocap_synced(:,2);
    
    %need to sort these depending on how the excel file was saved
    m_head = mocap_synced(:,I-1:I+1);
    
    m_rim_top = mocap_synced(:,RimTop-1:RimTop+1);
    m_axle = mocap_synced(:,Center-1:Center+1);

end

File: Wheel_Deflection/Python/getUsefulMarkers.py
"""
Get the indecies of the columns that pertain to the specific markers...
MTS head, top of the rim, and the center of the axel. Could be added upon
to include other points of interst.

Input:
mocap_synced = trimmed data from the motion capture
i = index of the MTS head column
rim_top = the index of the rimtop y-axis indicator
center = the index of the center y- axis indicator

Output:
mo_time = the time column from the mocap_synced
m_head = the columns pretaining to the head indicator
m_rim_top = the columns pretaining to the rim top indicator
x_axle = the columns pretaining to the axel indicator
"""


def get_useful_markers(mocap_synced, i, rim_top, center):
    mo_time = mocap_synced.iloc[:, 2]

    m_head = mocap_synced.iloc[:, i-1:i+1]

    m_rim_top = mocap_synced.iloc[:, rim_top-1:rim_top+1]
    m_axle = mocap_synced.iloc[:,center-1:center+1]

    return mo_time, m_head, m_rim_top, m_axle
